### ISSUE TITLE
Plain arrays are serialized (normalized) as "objects", ignoring serializeNull

### ISSUE THREAD

**Issue by scaytrase** [schmittjoh/serializer#641]
As soon as I have no response in original closed issue #571 , I've open a new one

Given commit https://github.com/schmittjoh/serializer/commit/b0a46c80cf3dbd97205f2e2ffeb9229080899c76 does not solve the problem - plain arrays are still serialized as object:

`test/JMS/Serializer/Tests/Serializer/ArrayTest.php`

```php
    public function testNullArraySerialization()
    {
        $context = SerializationContext::create();
        $context->setSerializeNull(true);
        $array = [0, null, 'a'];
        self::assertSame($array, $this->serializer->toArray($array, $context));
    }
```

```
Failed asserting that Array &0 (
    0 => 0
    2 => 'a'
) is identical to Array &0 (
    0 => 0
    1 => null
    2 => 'a'
).
```

How we force an array being normalized as array, not object? I'll try to file this failing test as pull request if you need it.

As a solution I propose to make this check 
https://github.com/schmittjoh/serializer/blob/master/src/JMS/Serializer/GenericSerializationVisitor.php#L101
look like others 
https://github.com/schmittjoh/serializer/blob/master/src/JMS/Serializer/YamlSerializationVisitor.php#L84-L85

**Comment by goetas** [comment]
related to  https://github.com/schmittjoh/serializer/pull/626 ?

**Comment by scaytrase** [comment]
Yes, thanks. I haven't re-check new issues about this since my closed #571

**Comment by goetas** [comment]
related to #626
